fabfile.py

```diff
@@ -208,22 +208,37 @@ def docker_compose_run(command_name, service="builder", user="app", no_deps=Fals
 
 def set_local_configuration():
     env.compose_files = ['docker-compose.yml']
-    env.user_id = int(local('id -u', capture=True))
-    env.root_dir = os.path.dirname(os.path.abspath(__file__))
+    env.dinghy = False
+    env.powerShell = False
+    env.user_id = 1000
+
+    if platform == "darwin":
+        env.dinghy = True
+    elif platform in ["win32", "win64"]:  # TODO Find a better check, for PowerShell versus cmd
+        env.powerShell = True
+        # TODO this is a mess. Python can't set the vars correctly on PowerShell
+        # and local() always calls cmd.exe!
+        shellProjectName = local('echo %PROJECT_NAME%', capture=True)
+        if (shellProjectName != env.project_name):
+            print 'Did you run the env vars?'
+            print '$Env:PROJECT_NAME="%s"' % env.project_name
+            print '$Env:PROJECT_DIRECTORY="%s"' % env.project_directory
+            print '$Env:PROJECT_HOSTNAMES="%s"' % env.project_hostnames
+            raise SystemError('Env vars not set (Windows detected)')
+
+    if not env.powerShell:
+        env.user_id = int(local('id -u', capture=True))
 
     if env.user_id > 256000:
         env.user_id = 1000
 
+    env.root_dir = os.path.dirname(os.path.abspath(__file__))
+
     with quiet():
         try:
             docker_kernel = "%s" % local('docker version --format "{{.Server.KernelVersion}}"', capture=True)
         except:
             docker_kernel = ''
 
-    if platform == "linux" or platform == "linux2" or docker_kernel.endswith('linuxkit-aufs'):
```

Comment:
Hum I wonder whether the check of `docker_kernel.endswith('linuxkit-aufs')` is still useful? Could this happen when platform is darwin (with docker for mac?)

Reply:
Not a blocker for this PR.

Reply:
Investigations points that the docker_kernel contains:

- `4.15.0-72-generic` on a Linux native Docker
- `4.14.98-boot2docker` on Dinghy
- `todo` on Docker For Mac
- `todo` on Docker For Windows